src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int GROWTH_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this.buckets = new Node[INITIAL_CAPACITY];
+        this.threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndexByKey(key);
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (keyEquals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        addNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndexByKey(key);
+        Node<K, V> currentNode = buckets[index];
+        while (currentNode != null) {
+            if (keyEquals(currentNode.key, key)) {
```

Comment:
this logic repeats 2 times
we can create a private method that returns node by key

and reuse it here (get value of this returned node), and in put() - if node is returned, replace value